### PR TITLE
Add Karpenter Node Instance Profile Name as a Cfn Output

### PR DESCRIPTION
**Pull request by youngjeong46**
*Issue #, if available:*
#494 

*Description of changes:*
PR addresses:
- Currently, the Karpenter Add-on does not export Instance Profile in any way. This is necessary for users to add as part of provisioners. This PR will expose the name of the Instance Profile as a CloudFormation output.
- Separate Karpenter Addon suite of unit tests
- Consolidation check fixed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/karpenter/index.ts [hunk 1]
```diff
@@ -5,9 +5,11 @@ import { ClusterInfo } from '../../spi';
 import { conflictsWith, createNamespace, createServiceAccount, dependable, setPath, } from '../../utils';
 import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from '../helm-addon';
 import { KarpenterControllerPolicy } from './iam';
+import { CfnOutput } from 'aws-cdk-lib';
 import * as md5 from 'ts-md5';
 import * as semver from 'semver';
 import * as assert from "assert";
+import { Instance } from 'aws-cdk-lib/aws-ec2';
```
Comment (shapirov103) [inline lib/addons/karpenter/index.ts:12]: (minor): please fix this unused import warning.

**Review by shapirov103 — COMMENTED** [review]
LGTM, minor comment

File: lib/addons/karpenter/index.ts (hunk 1 above)
Comment (youngjeong46) [reply lib/addons/karpenter/index.ts]: addressed

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**